src/utils/wallet.js

```diff
@@ -211,6 +214,34 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network) {
+    const metadatas = {};
+
+    for (const token of tokens) {
+      try {
+        const response = await metadataApi.getDag(token, network);
```

Comment:
Use `Promise.all` so it loads faster.
Do you think it's better to create a get method to a list of tokens?

Reply:
As we discussed privately it's better not to do it right now